### ISSUE TITLE
Building a Release tarball

### ISSUE THREAD

**Issue by kgilmer** [yshui/compton#224]
Hello, I would like to build your v7 release.  The README mentions that git submodules need to be initialized (to pull in dependencies I'm assuming) but since it's a release tarball there is no git metadata.  Is there alternate instructions that will work for releases?

**Comment by yshui** [comment]
@kgilmer You will have to manually clone https://github.com/yshui/test.h into `subprojects/test.h`

**Comment by kgilmer** [comment]
That fixed me up, thank you @yshui .  Would you be interested in a minor PR to add this and the dependency list in Ubuntu/Debian format to the readme?

**Comment by yshui** [comment]
@kgilmer Thanks, but I am planning to stop including `test.h` as a submodule in next release. Instead, it will be vendored as part of the compton repository. That should solve the release tarball problem.

Will close this as soon as that is done.

**Comment by kgilmer** [comment]
Ok, sounds good @yshui .  Well, here are the Ubuntu deps in case anyone comes across this in the future:
```
libxext-dev libxcb1-dev libxcb-damage0-dev libxcb-xfixes0-dev libxcb-shape0-dev libxcb-render-util0-dev libxcb-render0-dev libxcb-randr0-dev libxcb-composite0-dev libxcb-image0-dev libxcb-present-dev libxcb-xinerama0-dev libpixman-1-dev libdbus-1-dev libconfig-dev libxdg-basedir-dev libgl1-mesa-dev  libpcre2-dev  libevdev-dev uthash-dev libevdev2
```

**Comment by yshui** [comment]
Thanks, I will add that to the README

**Comment by kgilmer** [comment]
Glad it might be useful.  @yshui would you consider your v7 release, built with default flags, to be suitable to ship with a desktop environment?  I've been using it with https://regolith-linux.org for a few days now and haven't had any problems but thought I'd ask.

**Comment by yshui** [comment]
@kgilmer it's already [part](https://www.archlinux.org/packages/community/x86_64/compton/) of Arch Linux. Don't know if that's good enought for you :)

**Comment by kgilmer** [comment]
It certainly is. Thank you @yshui!

**Comment by wxl** [comment]
@kgilmer you are a godsend. I just started sitting down to package this for Ubuntu and started out with Ye Olde Compton's depends and they're nearly all wrong. You saved me a ton of time. Only thing missing is `libev-dev`. Perhaps you already had it installed.

@yshui call me crazy but I was thinking of maybe sneaking this into 19.10, which is due in October. You think you'll have a release to pull `test.h` into the `compton` repo relatively soon?

**Comment by yshui** [comment]
@wxl Will do a release with `test.h` today or next week.

**Comment by wxl** [comment]
@yshui that sounds great. Thanks a ton! If all goes well, not only will it be in the Ubuntu archive but Lubuntu will be shipping it. Squee!

**Comment by kgilmer** [comment]
Hi @wxl , glad to help and thanks for the additional packages.  I'll use those if I opt to package this.  But, if you're working on it maybe I don't have to.  If you packaged, this, would it be available before the 19.10 release?

**Comment by wxl** [comment]
@kgilmer I've got a functional but certainly not finished (I need to clean up some of the packaging) package in a [PPA](https://launchpad.net/~wxl/+archive/ubuntu/testy-testy) that you're welcome to test out. If all goes well, it will be in 19.10 before it's released, but I'm not currently targeting 19.04 or 18.04. I'm considering 18.04, but 19.04 is end of life in January, so it seems like it's not worth the effort. If you're curious, [here](https://phab.lubuntu.me/T96)'s where we're tracking this in Lubuntu.

**Comment by yshui** [comment]
Closed by release v7.3
